client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelFactory.java

```diff
@@ -282,6 +282,11 @@ public Map<SocketAddress, Set<Integer>> getPrimarySegmentsByAddress(byte[] cache
 
    public <T extends ChannelOperation> T fetchChannelAndInvoke(Set<SocketAddress> failedServers, byte[] cacheName,
                                                                T operation) {
+      return fetchChannelAndInvoke(failedServers, cacheName, operation, true);
+   }
+
+   private <T extends ChannelOperation> T fetchChannelAndInvoke(Set<SocketAddress> failedServers, byte[] cacheName,
```

Comment:
I do not like this change, many operations have no notion of being tied to a cache.

Reply:
Updated.